### PR TITLE
add support for replaceById operation

### LINKED ISSUE #48: Implement support for replaceById operation
`Error: The connector elasticsearch does not support replaceById operation. This is not a bug in LoopBack. Please contact the authors of the connector, preferably via GitHub issues.`

Get rid of this error when `POST /Model/{id}/replace` Replace attributes for a model instance and persist it into the data source.`

### PR DESCRIPTION
**Pull request by juanpujol**
This fixes the exeption when running: `POST /Model/{id}/replace`

`Error: The connector elasticsearch does not support replaceById
operation. This is not a bug in LoopBack. Please contact the
authors of the connector, preferably via GitHub issues.`

closes #48

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/esConnector.js [hunk 1]
```diff
@@ -939,7 +939,8 @@ ESConnector.prototype.destroy = function destroy(modelName, id, done) {
  * NOTES:
  * > The _source field need to be enabled for this feature to work.
  */
-ESConnector.prototype.updateAttributes = function updateAttrs(modelName, id, data, callback) {
+
+function updateAttrs(modelName, id, data, options, callback) {
```
Comment (juanpujol) [inline lib/esConnector.js]: Adding the options parameter fixes the callback hanging on loopback 3 alpha.

**Comment by bajtos** [comment]
@juanpujol thank you for the pull request!

I am personally not familiar with ES to judge whether this is a correct implementation.

The difference between `updateAttributes` and `replaceById` is that the first operation should perform a partial update (patch only the properties provided in the request, leave other properties untouched), while the second operation should replace the whole instance.

```
current instance:
{ "name": "Miroslav", "handle": "bajtos" }

updateAttributes({"name": "Miroslav Bajtos"})
  results in { "name": "Miroslav Bajtos", "handle": "bajtos" }

replaceById({"name": "Alter Ego" })
  results in { "name": "Alter Ego" }
  (handle was removed)
```

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by juanpujol** [comment]
@bajtos thank you, that's a good observation.

Just updated the PR to use the `index` API when calling `replaceById` that should take care of the scenario you described.

File: lib/esConnector.js (hunk 1 above)
Comment (pulkitsinghal) [inline lib/esConnector.js]: @juanpujol - if you agree with what you see below then can you also commit the following for backward compatibility:

```
// pseudo-code:
// if options is a function (not an object) then set `callback=options` and `options=null`
```

let me know

**Comment by pulkitsinghal** [comment]
@juanpujol - would you be able to contribute some tests as well? I can get you started via a skype session or something similar. PM me on gitter.im for my skype handle

**Pull request closed without merging**